core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -1822,6 +1822,13 @@ final class NameResolverErrorHandler implements Runnable {
         @Override
         public void run() {
           handleErrorInSyncContext(error);
+          ManagedChannelServiceConfig effectiveServiceConfig;
```

Comment:
You can merge these 2 lines.

Reply:
@shivaspeaks The above condition which you are asking us to merge is taken from the onResult() from the same class.

else if (defaultServiceConfig != null) {
              effectiveServiceConfig = defaultServiceConfig;
              realChannel.updateConfigSelector(effectiveServiceConfig.getDefaultConfigSelector());
              channelLogger.log(ChannelLogLevel.INFO,"Received no service config, using default service config");
}
So we can chnage this condition as below as per your comment.

if (defaultServiceConfig != null) {
  realChannel.updateConfigSelector(defaultServiceConfig.getDefaultConfigSelector());
  channelLogger.log(ChannelLogLevel.ERROR,"Initial Name Resolution error, using default service config");
}

Hope this change is fine? Please confirm.

Reply:
Yes, I've checked onResult() method. Over there, there are multiple cases where 'effectiveServiceConfig' variable was needed to be changed as per conditions, but here we are just using 'defaultServiceConfig' and we do not have any plans to change its value.

So having 
```
ManagedChannelServiceConfig effectiveServiceConfig;
effectiveServiceConfig = defaultServiceConfig;
```
is redundant.

Also, from merging those two lines I meant
```
ManagedChannelServiceConfig effectiveServiceConfig = defaultServiceConfig;
```

But anyways, it looks good to me.

Reply:
@shivaspeaks can we have quick call on meet please?

Reply:
@kannanjgithub @shivaspeaks Fixed the Junit case and it's going to our change/flow now. We are not able to Mock the Status with ArgumentCaptor as its a final class so as of now we are verifying the Status and not doing any assertions.
Even I have seen the existing UT cases with only verification and no assertion, is it fine? Please suggest further?

Reply:
How did you verify it? Can you add few assert statements that it indeed uses default service config that you provided from test? We should be able to know by assert statements success or failure. The unit test should itself test the behaviour of the function onError.

Please go through [this](https://github.com/grpc/grpc-java/blob/0fea7dd32e259f2d4abc2404115ad386d89e325d/core/src/test/java/io/grpc/internal/ManagedChannelImplTest.java#L3142) for onSuccess. Perhaps in a similar way, we can think of for onError as well.

Also a request to @Alam-Saran and others, considering [this comment](https://github.com/grpc/grpc-java/pull/11368#discussion_r1687521519) please do not do major edits to the comment after getting a response on it as we might lose the context of the thread thereafter. You can respond with a new comment with updates, that will be great.

Reply:
@shivaspeaks As per your feedback/suggestion added the assert statement.